### ISSUE TITLE
# in HTML String for WebView breaks the HTML Page

### ISSUE THREAD

**Issue by omaRaad** [nhn/gpm.unity#473]
## Service

* [x] WebView

## Version
2.2.5

## Describe the bug
When using ShowHtmlString, strings that include a `#` will get encoded causing the WebWiew to fail loading the remaining html.

## To Reproduce
1. Add a WebView to the project.
2. Create 
```c#
string HTML_STRING = @"<!DOCTYPE html> <html lang=""en""> <head> <meta charset=""utf-8"" /> <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' /> <style> html, body, #widget { height: 100%; margin: 0%; background: red; } </style> <title>Leaderboard</title> </head> <body> <div id=""widget""></div> </body> </html>";
```
3. Call 
```c#
GpmWebView.ShowHtmlString(
        HTML_STRING,
       new GpmWebViewRequest.Configuration()
       {
           style = GpmWebViewStyle.POPUP,
           backgroundColor = "#00ffffff",
           isClearCookie = true,
           isClearCache = true,
           position = GetPosition(),
           size = GetSize()
       },
       OnCallback,
       new List<string>()
       );
```
With OnCallback:
```c#
private void OnCallback(
    GpmWebViewCallback.CallbackType callbackType,
    string data,
    GpmWebViewError error)
    {
        Debug.Log("OnCallback: " + callbackType);
        switch (callbackType)
        {
            case GpmWebViewCallback.CallbackType.Open:
                if (error != null)
                {
                    Debug.LogFormat("Fail to open WebView. Error:{0}", error);
                }
                break;
            case GpmWebViewCallback.CallbackType.Close:
                if (error != null)
                {
                    Debug.LogFormat("Fail to close WebView. Error:{0}", error);
                }
                break;
            case GpmWebViewCallback.CallbackType.PageStarted:
                if (string.IsNullOrEmpty(data) == false)
                {
                    Debug.LogFormat("PageStarted Url : {0}", data);
                }
                break;
            case GpmWebViewCallback.CallbackType.PageLoad:
                if (string.IsNullOrEmpty(data) == false)
                {
                    Debug.LogFormat("Loaded Page: {0}", data);
                }
                else
                {
                    Debug.LogFormat("Fail to load Page. Error: {0}", error);
                }
                break;
            case GpmWebViewCallback.CallbackType.Scheme:
                if (error == null)
                {
                    if (data.Equals("USER_ CUSTOM_SCHEME") == true || data.Contains("CUSTOM_SCHEME") == true)
                    {
                        Debug.Log(string.Format("scheme:{0}", data));
                    }
                }
                else
                {
                    Debug.Log(string.Format("Fail to custom scheme. Error:{0}", error));
                }
                break;
            case GpmWebViewCallback.CallbackType.ExecuteJavascript:
                Debug.LogFormat("ExecuteJavascript data : {0}, error : {1}", data, error);
                break;
        }
    }
```
4. The html code breaks, and the PageLoad callback prints the following:
```
Loaded Page:data:text/html,<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8" /> <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' /> <style> html, body, #widget%20{%20height:%20100%;%20margin:%200%;%20background:%20red;%20}%20%3C/style%3E%20%3Ctitle%3ELeaderboard%3C/title%3E%20%3C/head%3E%20%3Cbody%3E%20%3Cdiv%20id=%22widget%22%3E%3C/div%3E%20%3C/body%3E%20%3C/html%3E
```

## Expected behavior
The HTML Code is read normally and the PageLoad callback prints:
```
Loaded Page:data:text/html,<!DOCTYPE html> <html lang="en"> <head> <meta charset="utf-8" /> <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no' /> <style> html, body, #widget { height: 100%; margin: 0%; background: red; } </style> <title>Leaderboard</title> </head> <body> <div id="widget"></div> </body> </html>
```

### Unity
- 2022.3

### Smartphone (please complete the following information):
 - Device: Samsung Galaxy Note9
 - OS: Android 10
 - GmpWebview

## Additional context
Printing the `HTML_STRING` outside of the GmpWebView shows it correctly.

**Comment by taehyeon-nhn** [comment]
Hello. @omaRaad.

We'll check it out and let you know in the comments.

Thank you.

**Comment by omaRaad** [comment]
Any Updates @taehyeon-nhn 
Thanks

**Comment by omaRaad** [comment]
@spin-nhn I would appreciate it if you give me an update. Thanks

**Comment by taehyeon-nhn** [comment]
Hello. @omaRaad.

We'll check the issue and let you know what we've fixed in the comments.

Thank you.

**Comment by omaRaad** [comment]
@taehyeon-nhn Any update about this? It's blocking my workflow so I would appreciate an update

**Comment by spin-nhn** [comment]
Hello. @omaRaad .

I bring you good news.
The same issue was found and the internal code was modified.
We will reflect related fixes in the next release.

Thank you.

@taehyeon-nhn @SangYun-nhn

**Comment by omaRaad** [comment]
@spin-nhn Is there any ETA for the next release? Or any board where I can check the progress?

Thanks

**Comment by spin-nhn** [comment]
Hello. @omaRaad .

The issue has been fixed and GPMWebview v2.0.5 has been released today.

Thank you.
